File: app/src/main/java/com/yuriy/openradio/shared/view/list/MediaItemViewHolder.java
package com.yuriy.openradio.shared.view.list;

import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.xenione.libs.swipemaker.SwipeLayout;
import com.yuriy.openradio.mobile.view.list.BothSideCoordinatorLayout;

public final class MediaItemViewHolder extends RecyclerView.ViewHolder {

    /**
     * Title text view.
     */
    public final TextView mNameView;

    public final TextView mBitrateView;

    /**
     * Description text view.
     */
    public final TextView mDescriptionView;

    /**
     * Category image view.
     */
    public final ImageView mImageView;

    /**
     * Check box vew for the "Favorites" option.
     */
    public final CheckBox mFavoriteCheckView;

    public final SwipeLayout mForegroundView;

    public final ViewGroup mRoot;

    public final ImageButton mSettingsView;

    public MediaItemViewHolder(final View view, final int root_view_id, final int name_view_id,
                               final int description_view_id, final int img_view_id, final int favorite_view_id,
                               final int bitrate_view_id, final int settings_btn_view_id,
                               final int foreground_view_id) {
        super(view);
        mRoot = view.findViewById(root_view_id);
        mNameView = view.findViewById(name_view_id);
        mDescriptionView = view.findViewById(description_view_id);
        mImageView = view.findViewById(img_view_id);
        mFavoriteCheckView = view.findViewById(favorite_view_id);
        mBitrateView = view.findViewById(bitrate_view_id);
        mSettingsView = view.findViewById(settings_btn_view_id);
        mForegroundView = view.findViewById(foreground_view_id);
    }
}

File: app/src/main/java/com/yuriy/openradio/shared/view/list/MediaItemViewHolder.kt
/*
 * Copyright 2020 The "Open Radio" Project. Author: Chernyshov Yuriy
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.yuriy.openradio.shared.view.list

import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import android.widget.ImageButton
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.xenione.libs.swipemaker.SwipeLayout

class MediaItemViewHolder(view: View, root_view_id: Int, name_view_id: Int,
                          description_view_id: Int, img_view_id: Int, favorite_view_id: Int,
                          bitrate_view_id: Int, settings_btn_view_id: Int,
                          foreground_view_id: Int) : RecyclerView.ViewHolder(view) {
    /**
     * Title text view.
     */
    @JvmField
    val mNameView: TextView = view.findViewById(name_view_id)

    @JvmField
    val mBitrateView: TextView = view.findViewById(bitrate_view_id)

    /**
     * Description text view.
     */
    @JvmField
    val mDescriptionView: TextView = view.findViewById(description_view_id)

    /**
     * Category image view.
     */
    @JvmField
    val mImageView: ImageView = view.findViewById(img_view_id)

    /**
     * Check box vew for the "Favorites" option.
     */
    @JvmField
    val mFavoriteCheckView: CheckBox = view.findViewById(favorite_view_id)

    @JvmField
    val mForegroundView: SwipeLayout? = view.findViewById(foreground_view_id)

    @JvmField
    val mRoot: ViewGroup = view.findViewById(root_view_id)

    @JvmField
    val mSettingsView: ImageButton? = view.findViewById(settings_btn_view_id)
}
